File: app/src/main/java/com/anokmik/tripassistant/databinding/list/ObservableListAttachStateChangeListener.java
package com.anokmik.tripassistant.databinding.list;

import android.databinding.ObservableList;
import android.view.View;

public final class ObservableListAttachStateChangeListener implements View.OnAttachStateChangeListener {

    private final ObservableList observableList;

    private final ObservableList.OnListChangedCallback onListChangedCallback;

    public ObservableListAttachStateChangeListener(ObservableList observableList, ObservableList.OnListChangedCallback onListChangedCallback) {
        this.onListChangedCallback = onListChangedCallback;
        this.observableList = observableList;

        observableList.addOnListChangedCallback(onListChangedCallback);
    }

    @Override
    public void onViewAttachedToWindow(View view) {

    }

    @Override
    public void onViewDetachedFromWindow(View view) {
        observableList.removeOnListChangedCallback(onListChangedCallback);
    }

}

File: app/src/main/kotlin/com/anokmik/tripassistant/databinding/list/ObservableListAttachStateChangeListener.kt
package com.anokmik.tripassistant.databinding.list

import android.databinding.ObservableList
import android.view.View

class ObservableListAttachStateChangeListener<T>(
        private val observableList: ObservableList<T>,
        private val onListChangedCallback: ObservableList.OnListChangedCallback<out ObservableList<T>>
) : View.OnAttachStateChangeListener {

    init {
        observableList.addOnListChangedCallback(onListChangedCallback)
    }

    override fun onViewAttachedToWindow(view: View) {

    }

    override fun onViewDetachedFromWindow(view: View) {
        observableList.removeOnListChangedCallback(onListChangedCallback)
    }

}